### ISSUE TITLE
A possible reason of `block_num != 0` error

### ISSUE THREAD

**Issue by abitmore** [steemit/steem#89]
In `get_blockchain_synopsis()`, `low_block_num` is initialized with `last_non_undoable_block_num()`,  which is `head_block_num() - _undo_db.size()` (related code is [here](https://github.com/steemit/steem/blob/ae6d87cc00b685aadaa027745d25e4b59af7c190/libraries/app/application.cpp#L617) and [here](https://github.com/steemit/steem/blob/ae6d87cc00b685aadaa027745d25e4b59af7c190/libraries/chain/database.cpp#L1976-L1979)). When `head_block_num() < 15` (or so), all blocks will be also in `undo_db`, in this case `head_block_num()` == `_undo_db.size()`, so `low_block_num` is initialized with 0, which will cause a `block_num != 0` assertion failure at [Line 715] (https://github.com/steemit/steem/blob/ae6d87cc00b685aadaa027745d25e4b59af7c190/libraries/app/application.cpp#L715).

<s>A possible fix is to add `else low_block_num = 1;` after Line 698.</s>

Update: above fix doesn't work.

**Comment by abitmore** [comment]
Above fix doesn't work. I'm working on another fix.

To reproduce the issue: 
* run `./steemd -s seed.steemed.net:2001 --checkpoint='[10,"1111111111111111111111111111111111111111"]'`, the program will crash after block 10 is rejected.

 ```
libraries/chain/block_database.cpp:109: steemit::chain::block_id_type steemit::chain::block_database::fetch_block_id(uint32_t) const: Assertion `block_num != 0' failed.
Aborted (core dumped)
 ```

With the saved chain data, restarting without `--checkpoint` will also lead to a crash. `--replay` doesn't work in this case. `--resync` works sometimes (due to race conditions).

**Comment by abitmore** [comment]
Fix submitted (see pull request #95).
